engines/sci/engine/script_patches.cpp

```diff
@@ -12304,6 +12306,171 @@ static const uint16 sq4CdPatchBikerTimepodMessage[] = {
+	0x31, 0x1d,                         // bnt 1d [ state 12 ]
+	PATCH_ADDTOOFFSET(+29),
+	0x3c,                               // dup
+	0x35, 0x0c,                         // ldi 0c
+	0x1a,                               // eq?
+	0x31, 0x34,                         // bnt 34 [ state 13 ]
+	0x38, PATCH_SELECTOR16(setCycle),   // pushi setCycle
+	0x78,                               // push1
+	0x76,                               // push0
+	0x7a,                               // push2 [ view ]
+	0x78,                               // push1
+	0x39, 0x0d,                         // pushi 0d
+	0x39, PATCH_SELECTOR8(loop),        // pushi loop
+	PATCH_ADDTOOFFSET(+32),
+	0x4a, 0x2a,                         // send 2a [ sp1 setCycle: 0 view: 13 loop: ... ]
+	PATCH_ADDTOOFFSET(+11),
+	0x31, 0x1e,                         // bnt 1e [ state 14 ]
+	PATCH_ADDTOOFFSET(+30),
+	0x3c,                               // dup
+	0x35, 0x0e,                         // ldi 0e
+	0x1a,                               // eq?
+	0x31, 0x34,                         // bnt 34 [ state 15 ]
+	0x38, PATCH_SELECTOR16(setCycle),   // pushi setCycle
+	0x78,                               // push1
+	0x76,                               // push0
+	0x7a,                               // push2 [ view ]
+	0x78,                               // push1
+	0x39, 0x0d,                         // pushi 0d
+	0x39, PATCH_SELECTOR8(loop),        // pushi loop
+	PATCH_ADDTOOFFSET(+37),
+	0x4a, 0x2c,                         // send 2c [ sp2 setCycle: 0 view: 13 loop: ... ]
+	PATCH_END
+};
+
+// Room 370 in front of Sock's is missing a flag check and runs a lethal Sequel
+//  Police script when escaping the Skate-O-Rama after hiding in the electronics
+//  store. A lockup bug in the CD version prevented getting this far. As this is
+//  the wrong time to run the script, the police are drawn as if they're still
+//  swimming in zero gravity.
+//
+// Flag 23 is set when the police pursue in escalator room 400 and triggers the
+//  squeeze scripts that force entering the Skate-O-Rama in rooms 370 and 390.
+//  Flag 22 is set when exiting the Skate-O-Rama after dodging the police. The
+//  two rooms handle these flags differently and are out of sync. Room 370
+//  clears flag 23 and doesn't test 22 while room 390 doesn't clear 23 and does
+//  test 22. The result is that room 370 doesn't see that you've escaped the
+//  Skate-O-Rama and assumes the police are still pursuing from room 400.
+//
+// We fix this by only running sp1Squeeze in room 370 when coming from room 400.
+//  This is equivalent to adding the missing flag 22 check but can be done in
+//  the available bytes. To make room we use the clear-flag method's return
+//  value, which is the previous flag value, and only test if the previous room
+//  number is even, which only room 400 is. This patch is split into two parts
+//  to surround a single bnt instruction whose size changed between versions.
```

Comment:
Sorry, I'm confused - can't you create a single patch in this case?

Reply:
The bnt instruction that's in the middle has a 16-bit operand in some versions and an 8-bit in the other (even though it's the same value) and there's not a mechanism for "skip 2 or 3 bytes".

I updated the comment to be clearer that it's the operand size in the instruction.

Reply:
That makes sense. Thanks!